libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.h

```diff
@@ -177,6 +177,7 @@ class HTTPClient
     void setAuthorization(const char * user, const char * password);
     void setAuthorization(const char * auth);
     void setTimeout(uint16_t timeout);
+    bool setProxyHost(const String &proxyUrl);
```

Comment:
Sample usage 

```
WiFiClient client;
HTTPClient http;

http.setProxyHost("http://192.168.1.1:8080");
http.begin(client, "http://foo.bar/");
  
// If the proxy requires auth
http.addHeader("Proxy-Authorization", "Basic aaaaaaaaaaaa=");
http.GET();
```

Reply:
sure, can you point me to where those are located?

Reply:
Here's the HTTPClient [examples location](https://github.com/esp8266/Arduino/tree/master/libraries/ESP8266HTTPClient/examples).
As you can see, the file hierarchy is `myExampleName/myExampleName.ino`.

Documentation is ...
... there's absolutely no official documentation on `HTTPClient`.
So your addition will be added in the documentation when it will be written for the whole class, or if we add doxygen at some point.